dll/win32/browseui/internettoolbar.cpp

```diff
@@ -1727,9 +1715,8 @@ LRESULT CInternetToolbar::OnContextMenu(UINT uMsg, WPARAM wParam, LPARAM lParam,
     SHCheckMenuItem(contextMenu, IDM_TOOLBARS_ADDRESSBAR, IsBandVisible(ITBBID_ADDRESSBAND) == S_OK);
     SHCheckMenuItem(contextMenu, IDM_TOOLBARS_LINKSBAR, FALSE);
     SHCheckMenuItem(contextMenu, IDM_TOOLBARS_CUSTOMIZE, FALSE);
-    SHCheckMenuItem(contextMenu, IDM_TOOLBARS_LOCKTOOLBARS, fLocked);
-    goButtonChecked = SHRegGetBoolUSValueW(L"Software\\Microsoft\\Internet Explorer\\Main", L"ShowGoButton", FALSE, TRUE);
-    SHCheckMenuItem(contextMenu, IDM_TOOLBARS_GOBUTTON, goButtonChecked);
+    SHCheckMenuItem(contextMenu, IDM_TOOLBARS_LOCKTOOLBARS, gSettings.fLocked);
```

Comment:
If one clicks on the "lock toolbars" in one explorer window, this appears to retro-affect how the corresponding menu item is going to be shown **in other already-existing opened windows** (once one opens the corresponding menu). Is this a behaviour that also happens on Windows?

Reply:
This does not affect existing windows because it only reads the registry when initialized. This is the same behavior as Windows Server 2003.

Reply:
I mean, the check-menu-item call looks now at the global `gSettings.fLocked` .... So if another window has already deselected this item, it seems now that for the current window, opening again the menu would show the menu item deselected as well, even if that current window still has its toolbars still shown in the old state (and same for the status bar too).
See picture below:

![image](https://github.com/reactos/reactos/assets/1969829/9f28fe7a-616d-4c9e-be6c-dde336ecced0)

In addition (not shown here), when navigating to a new folder (in the window "2" above), the status bar will disappear (as discussed in another comment of ours below), *but* the toolbars stay in their original state (here, locked).

Reply:
Since each browseui window has its own copy of the settings struct and it doesn't reload its settings on navigation; this implementation allows the other window to keep its setting for the status bar and locked toolbar state. It does not cause a mismatch between the checked state of either item and the actual UI in that window.

Reply:
How is that each window has its own copy of the settings struct? Aren't they all using the same global gSettings variable?

Reply:
I see what you are saying now. My bad.

Reply:
> but the toolbar state does not refresh

Also note that when I opened the context menu in window 2, it still shows the menu item that says "Lock tool bars" being enabled (checked) , contrary to what's in window 1.
Which makes me tell that your usages of `gSettings.fLocked`, for example this `SHCheckMenuItem(contextMenu, IDM_TOOLBARS_LOCKTOOLBARS, gSettings.fLocked);` call above (and some others down below), look "weird" and should instead use a per-window `fLocked` variable instead. (Which is of course initially initialized with the value of `gSettings.fLocked`, but then that works independently.)

Reply:
If we are really concerned about this, I think the best way is to use a local settings struct for shellbrowser and get rid of the `RefreshGlobalUI` hack.

Reply:
I don't think a separate `fLocked` variable is a good idea. The purpose of this PR is to make it easier to add setting persistence to BrowseUI. Maybe what we really need is some sort of refresh message when `gSettings.save()` is called.